src/librustc_resolve/error_codes.rs

```diff
@@ -1660,6 +1660,20 @@ fn const_id<T, const N: T>() -> T { // error: const parameter
 }
 ```
 "##,
+
+E0734: r##"
+Type parameter defaults cannot use `Self` on structs, enums, or unions.
+Erroneous code example:
```

Comment:
Please add an empty line before this one.

Reply:
done, thanks!